src/query/src/dummy_catalog.rs

```diff
@@ -168,12 +168,12 @@ impl TableProvider for DummyTableProvider {
             .collect();
         request.limit = limit;
 
-        let stream = self
+        let scanner = self
             .engine
-            .handle_query(self.region_id, request)
+            .handle_partitioned_query(self.region_id, request)
             .await
             .map_err(|e| DataFusionError::External(Box::new(e)))?;
-        Ok(Arc::new(StreamScanAdapter::new(stream)))
+        Ok(Arc::new(ReadFromRegion::new(scanner)))
```

Comment:
The struct name changes from a noun to a verb, do want to think it a better name?

Reply:
Naming is hard. Do you have suggestions?